### ISSUE TITLE
FHIR Subscription

### ISSUE THREAD

**Issue by miriamroessner** [samply/blaze#838]
[gelöscht, hat sich erledigt]

**Comment by sidharthramesh** [comment]
How can I set up a FHIR Subscription?

 I've tried `POST /Subscription` with:

```json
{
	"resourceType": "Subscription",
	"reason": "Send all encounter creates and updates",
	"status": "requested",
	"criteria": "Encounter?",
	"channel": {
		"type": "rest-hook",
		"endpoint": "https://requestinspector.com/inspect/01gm912arqgj0p72sxtzbs7xp9",
		"payload": "application/fhir+json"
	}
}

```

but this does not change the "status" to "active" when getting it back, and also does not send webhooks to the endpoint on making changes to the Encounter resources.

I've also tried this with `"status": "active"` but this also does not seem to be sending HTTP requests to the endpoint.

Any idea on how to enable subscriptions?

Thanks again for this great project! Probably the most performant open-source FHIR server available!

**Comment by miriamroessner** [comment]
Hi. As far as I can tell your resource is correct. The problem is that the Blaze FHIR Server does not support Subscriptions at the moment.

**Comment by sidharthramesh** [comment]
Alright. I assume it was implemented since you closed this issue. Can you re-open it?

**Comment by alexanderkiel** [comment]
Yes FHIR subscription resources are currently just stored like all other resource types. But I already started implementing subscriptions. It no small task to do but I'll continue to work on it.

**Comment by nilskre** [comment]
Hi @alexanderkiel ,
I'm currently exploring the capabilities of the Blaze FHIR server and would like to clarify the current status of FHIR Subscriptions support.
My use case is the following: Whenever specific observation FHIR resources for specific patients are created or updated, I want to trigger an external event.
The idea was to use FHIR subscriptions mechanism for this. So on my FHIR server I have configured the subscription and whenever one of the configured resources changes an external API is called.

Are such use cases possible with the current Blaze server implementation?

Thanks in advance!

**Comment by alexanderkiel** [comment]
Hi Nils, unfortunately FHIR Subscriptions are still not implemented. However on the database layer, I have that functionality since a year or so. So it would be not a big thing to implement the FHIR R4 Subscription web hooks. Do you have time to contribute or at least test the implementation?

**Comment by nilskre** [comment]
Thanks for your fast reply and great to hear, that on the database layer everything is already implemented.
Since I’m not familiar with Clojure, I wouldn’t be able to implement it myself in a reasonable timeframe.
However, I’d be more than happy to help in testing the implementation.
